lib/maths/CSampling.cc

```diff
@@ -191,45 +191,44 @@ void doCategoricalSampleWithoutReplacement(RNG& rng,
         return;
     }
 
-    std::size_t p = probabilities.size();
-    if (n >= p) {
-        result.assign(boost::counting_iterator<std::size_t>(0),
-                      boost::counting_iterator<std::size_t>(p));
+    std::size_t m{probabilities.size()};
+    result.assign(boost::counting_iterator<std::size_t>(0),
+                  boost::counting_iterator<std::size_t>(m));
+
+    if (n >= m) {
+        return;
     }
 
     // Construct the transform function.
-    for (std::size_t i = 1u; i < p; ++i) {
+    for (std::size_t i = 1; i < probabilities.size(); ++i) {
```

Comment:
In other places you switched from `probabilities.size()` to `m`, but here you've switched the other way.

Reply:
See [d957489](https://github.com/elastic/ml-cpp/pull/1312/commits/d957489ee19d6eedc001373c3fac354aec037986).